src/Services/ExtensionService.cs

```diff
@@ -204,6 +206,38 @@ public async Task<IEnumerable<IExtensionWrapper>> GetInstalledExtensionsAsync(bo
         }
     }
 
+    private async Task<IEnumerable<string>> GetInstalledWidgetExtensionsAsync()
+    {
+        await _getInstalledExtensionsLock.WaitAsync();

```

Comment:
I originally gave it its own lock but the wait never ended, I thought maybe the catalog opening was tripping it up. I'll try again though, since I did make significant changes after that.

Reply:
Working now, gave this it's own lock.